src/main/java/duke/NewTaskWindowController.java

```diff
@@ -0,0 +1,159 @@
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.control.*;
+import javafx.scene.text.Text;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+
+public class NewTaskWindowController {
+
+    @FXML
+    private TextArea descriptionTextField;
+    @FXML
+    private Button confirmButton;
+    @FXML
+    private ComboBox<String> typeSelector;
+    @FXML
+    private Label dateLabel;
+    @FXML
+    private DatePicker datePicker;
+    @FXML
+    private TextField hours;
+    @FXML
+    private TextField minutes;
+    @FXML
+    private Text dots;
+
+    private MainWindowController parentController;
+
+    private void hideTimeFields() {
+        hours.setVisible(false);
+        minutes.setVisible(false);
+        dots.setVisible(false);
+    }
+
+    private void showTimeFields() {
+        hours.setVisible(true);
+        minutes.setVisible(true);
+        dots.setVisible(true);
+    }
+
+    void setParentController(MainWindowController parentController) {
+        this.parentController = parentController;
+    }
+
+    @FXML
+    void confirm(ActionEvent event) {
+        TaskList taskList = parentController.getDuke().tasks;
+        Storage storage = parentController.getDuke().storage;
+        if (!typeSelector.getSelectionModel().isEmpty()) {
+            String selected = typeSelector.getSelectionModel().getSelectedItem();
+            try {
+                switch (selected) {
+                case "To-Do":
+                    taskList.addTask(new ToDo(descriptionTextField.getText()), storage);
+                    break;
+                case "Deadline": {
+                    int hrs = Integer.parseInt(hours.getText());
+                    int mins = Integer.parseInt(minutes.getText());
+                    if (hrs >= 24 || hrs < 0 || mins < 0 || mins >= 60) {
+                        throw new DukeException("Invalid time inputs.");
+                    }
+                    taskList.addTask(
+                            new Deadline(descriptionTextField.getText(),
+                                    datePicker.getValue().atTime(
+                                            hrs,
+                                            mins)),
+                            storage);
+                    break;
+                }
+                case "Event": {
+                    int hrs = Integer.parseInt(hours.getText());
+                    int mins = Integer.parseInt(minutes.getText());
+                    if (hrs >= 24 || hrs < 0 || mins < 0 || mins >= 60) {
+                        throw new DukeException("Invalid time inputs.");
+                    }
+                    taskList.addTask(
+                            new Event(descriptionTextField.getText(),
+                                    datePicker.getValue().atTime(
+                                            hrs,
+                                            mins)),
+                            storage);
```

Comment:
Good job on parameters indentation. Perhaps you can simplify the "datePicker......mins" though as it can seem a bit messy.